Submitted By:            Douglas R. Reno <renodr at linxufromscratch dot org>
Date:                    2024-11-05
Initial Package Version: 24.8.2.1
Origin:                  Self
Upstream Status:         Pending
Description:             Fixes building Libreoffice with ICU-76 by adding two
                         #defines before the ICU headers in Firebird, to allow
                         for more functions to be visible. This also fixes an
                         issue with libcdr by updating the list of libraries
                         that are needed to link to ICU.

diff -Naurp libreoffice-24.8.2.1.orig/external/firebird/firebird-icu76.patch libreoffice-24.8.2.1/external/firebird/firebird-icu76.patch
--- libreoffice-24.8.2.1.orig/external/firebird/firebird-icu76.patch	1969-12-31 18:00:00.000000000 -0600
+++ libreoffice-24.8.2.1/external/firebird/firebird-icu76.patch	2024-11-01 11:57:57.599782179 -0500
@@ -0,0 +1,13 @@
+diff -Naurp src.orig/common/unicode_util.cpp src/common/unicode_util.cpp
+--- src.orig/common/unicode_util.cpp	2020-10-20 03:40:05.000000000 -0500
++++ src/common/unicode_util.cpp	2024-11-01 11:55:44.683926741 -0500
+@@ -40,6 +40,9 @@
+ #include "../common/StatusHolder.h"
+ #include "../common/os/path_utils.h"
+ 
++#define U_SHOW_CPLUSPLUS_API 0
++#define U_SHOW_CPLUSPLUS_HEADER_API 0
++
+ #include <unicode/ustring.h>
+ #include <unicode/utrans.h>
+ #include <unicode/uchar.h>
diff -Naurp libreoffice-24.8.2.1.orig/external/firebird/UnpackedTarball_firebird.mk libreoffice-24.8.2.1/external/firebird/UnpackedTarball_firebird.mk
--- libreoffice-24.8.2.1.orig/external/firebird/UnpackedTarball_firebird.mk	2024-11-01 11:56:39.869283007 -0500
+++ libreoffice-24.8.2.1/external/firebird/UnpackedTarball_firebird.mk	2024-11-01 11:58:37.035034367 -0500
@@ -51,6 +51,7 @@ $(eval $(call gb_UnpackedTarball_add_pat
     external/firebird/0001-extern-cloop-Missing-dependencies-of-compilations-on.patch.1 \
     external/firebird/configure-c99.patch \
     external/firebird/Wincompatible-function-pointer-types.patch \
+	external/firebird/firebird-icu76.patch \
     external/firebird/c++26.patch \
     external/firebird/c++20.patch \
 ))
diff -Naurp libreoffice-24.8.2.1.orig/external/libcdr/ExternalProject_libcdr.mk libreoffice-24.8.2.1/external/libcdr/ExternalProject_libcdr.mk
--- libreoffice-24.8.2.1.orig/external/libcdr/ExternalProject_libcdr.mk	2024-11-01 11:56:39.874283039 -0500
+++ libreoffice-24.8.2.1/external/libcdr/ExternalProject_libcdr.mk	2024-11-01 12:13:14.743047028 -0500
@@ -27,6 +27,7 @@ $(call gb_ExternalProject_get_state_targ
 	$(call gb_Trace_StartRange,libcdr,EXTERNAL)
 	$(call gb_ExternalProject_run,build,\
 		export PKG_CONFIG="" \
+		export ICU_LIBS="-licui18n -licuuc -licudata" \
 		&& unset MSYS_NO_PATHCONV && MAKE=$(MAKE) $(gb_RUN_CONFIGURE) ./configure \
 			--with-pic \
 			--enable-static \
